gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -143,25 +146,23 @@ def create_per_sample_counts_ht(
         )
     if rare_variants:
         filter_expr["rare_variants"] = mt.freq[0].AF < rare_variants_af
-    if by_csqs and not pass_filters:
-        raise Warning(
-            "Variant consequences only calculated in regions which pass filters..."
-        )
-    if by_csqs & pass_filters:
-        for csq in CSQ_SET:
-            is_lof = csq in LOF_CSQ_SET
-            csq_str = f"pass_filters_{'lof_' if is_lof else ''}{csq}"
-            filter_expr[csq_str] = (mt.most_severe_csq == csq) & (
-                filter_expr["pass_filters"]
+    if by_csqs:
+        filter_expr["lof"] = (
+            filter_expr["pass_filters"]
+            & hl.if_else(
+                hl.any(lambda csq: mt.most_severe_csq == csq, LOF_CSQ_SET), True, False
             )
-            if is_lof:
-                filter_expr[csq_str] = (filter_expr[csq_str]) & (~hl.is_missing(mt.lof))
-            if rare_variants:
-                filter_expr[f"rare_vairants_{csq_str}"] = (
-                    filter_expr[csq_str] & filter_expr["rare_variants"]
-                )
+            & (mt.lof == "HC")
+            & mt.no_lof_flags
+        )
+        filter_expr["missense"] = filter_expr["pass_filters"] & (
+            mt.most_severe_csq == "missense_variant"
+        )
+        filter_expr["synonymous"] = filter_expr["pass_filters"] & (
+            mt.most_severe_csq == "synonymous_variant"
+        )
```

Comment:
everything being pass_filters is nice

Reply:
I went through Konrad's code, wanted to apply filter_low_conf_regions, and passing filters and only canonical for these consequences. He put "HC" as the first filter, I didn't want to do the sequential counts yet.